packages/cloudflare-worker/src/auth.ts

```diff
@@ -9,75 +23,338 @@ export type AuthOptions = {
 export type AuthHeader = string | null | undefined
 
 export function initAuth(opts: AuthOptions) {
-    const nodeVersion = initBaseAuth({
-        authUrl: opts.authUrl,
-        apiKey: opts.apiKey,
-        manualTokenVerificationMetadata: {
-            verifierKey: opts.verifierKey,
-            issuer: opts.authUrl,
-        },
-    })
+    const authUrl: URL = validateAuthUrl(opts.authUrl)
+    const integrationApiKey: string = opts.apiKey
+    const apis = getApis(authUrl, integrationApiKey)
+
+    const publicKeyPromise = jose.importSPKI(opts.verifierKey, "RS256")
+
+    const validateAccessTokenAndGetUserClass = wrapValidateAccessTokenAndGetUserClass(publicKeyPromise, authUrl.origin)
+    const validateAccessTokenAndGetUser = wrapValidateAccessTokenAndGetUser(publicKeyPromise, authUrl.origin)
+    const validateAccessTokenAndGetUserWithOrgInfo = wrapValidateAccessTokenAndGetUserWithOrgInfo(
+        publicKeyPromise,
+        authUrl.origin
+    )
+    const validateAccessTokenAndGetUserWithOrgInfoWithMinimumRole =
+        wrapValidateAccessTokenAndGetUserWithOrgInfoWithMinimumRole(publicKeyPromise, authUrl.origin)
+    const validateAccessTokenAndGetUserWithOrgInfoWithExactRole =
+        wrapValidateAccessTokenAndGetUserWithOrgInfoWithExactRole(publicKeyPromise, authUrl.origin)
+    const validateAccessTokenAndGetUserWithOrgInfoWithPermission =
+        wrapValidateAccessTokenAndGetUserWithOrgInfoWithPermission(publicKeyPromise, authUrl.origin)
+    const validateAccessTokenAndGetUserWithOrgInfoWithAllPermissions =
+        wrapValidateAccessTokenAndGetUserWithOrgInfoWithAllPermissions(publicKeyPromise, authUrl.origin)
+
+    // Note: We exclude fetchTokenVerificationMetadata from the returned object
+    // because we have explicit usage of it above. Thus, it is not used in the returned object.
+    const { fetchTokenVerificationMetadata, ...nodeApis } = apis
```

Comment:
@andrew-propelauth I don't think this is relevant for this library -- only for node

Reply:
Ah yeah, in either case, we don't need to expose that function since it isn't relevant for users of this library either.